.github/actions/bpftrace/scripts/check-encryption-leaks.bt

```diff
@@ -359,25 +359,45 @@ kprobe:tcp_close
   }
 }
 
-// Trace UDP messages sent by the DNS proxy, even if the source address belongs to the host.
+// Trace UDP messages sent by the DNS proxy:
+// - IPSec:     skip packet if source address is CiliumInternalIP;
+// - Wireguard: always trace packet.
 kprobe:udp_sendmsg /comm == "cilium-agent" || comm == "dnsproxy"/
 {
   $sk = ((struct sock *) arg0);
   if (bswap($sk->__sk_common.skc_dport) == PORT_DNS) {
-    @trace_ip4[$sk->__sk_common.skc_rcv_saddr, bswap($sk->__sk_common.skc_num), PROTO_UDP] = SK_TRACK_FLAG_BASE;
-    @trace_sk[$sk] = true;
-    @sanity[TYPE_PROXY_DNS_IP4] = true;
+    $src_is_pod = (bswap($sk->__sk_common.skc_rcv_saddr) & MASK4) == CIDR4;
+    $src_is_internal =
+        $sk->__sk_common.skc_rcv_saddr == (uint32)pton(NODE1_INTERNAL_IPV4) ||
+        $sk->__sk_common.skc_rcv_saddr == (uint32)pton(NODE2_INTERNAL_IPV4) ||
+        $sk->__sk_common.skc_rcv_saddr == (uint32)pton(NODE3_INTERNAL_IPV4);
+
+    if (!(ENCRYPTION_TYPE_IPSEC && $src_is_pod && $src_is_internal)) {
+      @trace_ip4[$sk->__sk_common.skc_rcv_saddr, bswap($sk->__sk_common.skc_num), PROTO_UDP] = SK_TRACK_FLAG_BASE;
+      @trace_sk[$sk] = true;
+      @sanity[TYPE_PROXY_DNS_IP4] = true;
+    }
```

Comment:
> unless forced to with the apposite unused flag --use-cilium-internal-ip-for-ipsec

Yeah this can be a problem if we want to port check-leaks.bt to cloud env where --use-cilium-internal-ip-for-ipsec is enabled

Reply:
Mmm... what about another flag `USE_INTERNAL_FOR_CRYPTO` (args n.9) to skip those condition when set to true?

Reply:
yeah, in future. For now it's okay.

Reply:
I'll mark this as solved in the next round of push, where I'll add a reminder in the code.